Demo/app/src/main/java/com/juanocampo/swipeable/swipeablelist/swpeable/adapter/SwipeableHelperAdapter.java

```diff
@@ -75,18 +77,23 @@ public final void onBindViewHolder(final RecyclerView.ViewHolder holder, int pos
      * @see RecyclerView.ViewHolder#getAdapterPosition()
      */
     public void onItemDismiss(int position) {
-        listener.swiped(position);
+        int visiblePositionSelected = position - manager.findFirstVisibleItemPosition();
+        listener.swiped(visiblePositionSelected);
     }
 
-    private void restoreSwipedItem() {
-        final RecyclerView.ViewHolder viewHolder = swipeableViewHolders.get(callback.getLastSelectedPosition());
+    public void restoreSwipedItem(int restorePosition) {
+
+        if (swipeableViewHolders != null && !swipeableViewHolders.isEmpty() && restorePosition < swipeableViewHolders.size()) {
```

Comment:
rather than doing this, you can just pass in the viewholder directly by calling findViewHolderForAdapterPosition(lastSelectedPosition);